tuned/plugins/plugin_scheduler.py

```diff
@@ -136,8 +136,230 @@ def get_priority_max(self, sched):
+	for IRQs. The CPU affinity mask is adjusted for all processes and
+	threads matching [option]`ps_whitelist` option subject to success
+	of the `sched_setaffinity()` system call. The default setting of
+	the [option]`ps_whitelist` regular expression is `.*` to match all
+	processes and thread names. To exclude certain processes and threads
+	use [option]`ps_blacklist` option. The value of this option is also
+	interpreted as a regular expression and process/thread names (`ps -eo
+	cmd`) are matched against that expression. Profile rollback allows
+	all matching processes and threads to run on all CPUs and restores
+	the IRQ settings prior to the profile application.
+	+
+	Multiple regular expressions for [option]`ps_whitelist`
+	and [option]`ps_blacklist` options are allowed and separated by
+	`;`. Quoted semicolon `\;` is taken literally.
+	+
+	.Isolate CPUs 2-4
+	====
+	----
+	[scheduler]
+	isolated_cores=2-4
+	ps_blacklist=.*pmd.*;.*PMD.*;^DPDK;.*qemu-kvm.*
+	----
+	Isolate CPUs 2-4 while ignoring processes and threads matching
+	`ps_blacklist` regular expressions.
+	====
+	The [option]`default_irq_smp_affinity` option controls the values
+	*TuneD* writes to `/proc/irq/default_smp_affinity`. The file specifies
+	default affinity mask that applies to all non-active IRQs. Once an
+	IRQ is allocated/activated its affinity bitmask will be set to the
+	default mask.
+	+
+	The following values are supported:
+	+
+	--
+	`calc`::
+	Content of `/proc/irq/default_smp_affinity` will be calculated
+	from the `isolated_cores` parameter.  Non-isolated cores
+	are calculated as an inversion of the `isolated_cores`. Then
+	the intersection of the non-isolated cores and the previous
+	content of `/proc/irq/default_smp_affinity` is written to
+	`/proc/irq/default_smp_affinity`.  If the intersection is
+	an empty set, then just the non-isolated cores are written to
+	`/proc/irq/default_smp_affinity`.  This behavior is the default if
+	the parameter `default_irq_smp_affinity` is omitted.
+	`ignore`::
+	*TuneD* will not touch `/proc/irq/default_smp_affinity`.
+	explicit cpulist::
+	The cpulist (such as 1,3-4) is unpacked and written directly to
+	`/proc/irq/default_smp_affinity`.
+	--
+	+
+	.An explicit CPU list to set the default IRQ smp affinity to CPUs 0 and 2
+	====
+	----
+	[scheduler]
+	isolated_cores=1,3
+	default_irq_smp_affinity=0,2
+	----
+	====
+	To adjust scheduling policy, priority and affinity for a group of
+	processes/threads, use the following syntax.
+	+
+	[subs="+quotes,+macros"]
+	----
+	group.__groupname__=__rule_prio__:__sched__:__prio__:__affinity__:__regex__
+	----
+	+
+	where `__rule_prio__` defines internal *TuneD* priority of the
+	rule. Rules are sorted based on priority. This is needed for
+	inheritence to be able to reorder previously defined rules. Equal
+	`__rule_prio__` rules should be processed in the order there were
+	defined, however, this is Python interpreter dependant. To disable
+	an inherited rule for `__groupname__` use:
```

Comment:
Equal `__rule_prio__` rules should be processed in the order they were defined. However, this is Python interpreter dependant.